pkg/lib/oauth/token_encoding.go

```diff
@@ -53,6 +61,30 @@ func (e *AccessTokenEncoding) EncodeAccessToken(client *config.OAuthClientConfig
 	// verified JWT.
 	_ = claims.Set(jwt.JwtIDKey, grant.TokenHash)
 
+	claimsMap, err := jwtutil.ToMap(claims)
+	if err != nil {
+		return "", err
+	}
+
+	eventPayload := &blocking.UserSessionJWTPreCreateBlockingEventPayload{
+		UserRef: model.UserRef{
+			Meta: model.Meta{
+				ID: userID,
+			},
+		},
+		Payload: claimsMap,
+	}
+
+	err = e.Events.DispatchEvent(eventPayload)
+	if err != nil {
+		return "", err
+	}
+
+	claims, err = jwtutil.BuildFromMap(eventPayload.Payload)
```

Comment:
Instead of letting the user edit the jwt freely, should we restrict the user from editing some mandatory fields which will affect the jwt validity (e.g. `sub`, `iss`... etc)? Or the developer should take responsibility for that? 

Currently, if the hook returns a response that doesn't have those fields. The server will generate an invalid jwt. 

Open to discussion.